src/users/users.service.ts

```diff
@@ -108,95 +54,33 @@ export class UsersService {
   }
 
   async update(id: number, updateUserDto: UpdateUserDto) {
-    const user = await this.prismaService.user.findUnique({
-      where: { id },
-      select: { role: true },
-    });
-
-    if (!user) throw new NotFoundException();
-
-    const userData = { active: updateUserDto.active };
-
-    const userAttributesData = {
-      firstname: updateUserDto.firstname,
-      lastname: updateUserDto.lastname,
-      city: updateUserDto.city,
-      phone: updateUserDto.phone,
-      type: updateUserDto.type,
-    };
-
-    const organisationAttributesData = {
-      name: updateUserDto.name,
-      street: updateUserDto.street,
-      oib: updateUserDto.oib,
-      city: updateUserDto.city,
-    };
-
-    return this.prismaService.user.update({
-      where: { id },
-      data: {
-        ...userData,
-        ...(user.role === Role.USER && {
-          userAttributes: { update: { ...userAttributesData } },
-        }),
-        ...(user.role === Role.ORGANISATION && {
-          organisationAttributes: { update: { ...organisationAttributesData } },
-        }),
-      },
-      select: {
-        userAttributes: true,
-        organisationAttributes: true,
-        id: true,
-        email: true,
-        active: true,
-      },
-    });
+    return this.usersRepository.update(id, updateUserDto);
   }
 
   async remove(id: number) {
-    const user = await this.prismaService.user.findUnique({
-      where: { id },
-      select: {
-        userAttributes: { select: { certificates: { select: { key: true } } } },
-        email: true,
-      },
-    });
-    if (user?.userAttributes?.certificates?.length) {
-      const keys = user.userAttributes.certificates.map(({ key }) => key);
+    const certificates = await this.usersRepository.remove(id);
+
+    if (certificates?.length) {
+      const keys = certificates.map(({ key }) => key);
       await this.s3Service.deleteMany(keys);
     }
-
-    return this.prismaService.user.delete({ where: { id } });
   }
 
   async uploadProfilePhoto(
     id: number,
     file: Express.Multer.File,
   ): Promise<UploadProfilePhotoResponse> {
-    const user = await this.prismaService.user.findUniqueOrThrow({
-      where: { id },
-      select: {
-        id: true,
-      },
-    });
-
     const fileExt = extname(file.originalname);
-    const filename = createHash('md5').update(String(user.id)).digest('hex');
+    const filename = createHash('md5').update(String(id)).digest('hex');
     const profilePhotoKey = `profile_photos/${filename}${fileExt}`;
 
+    await this.usersRepository.uploadProfilePhoto(id, profilePhotoKey);
```

Comment:
@aarbanas We should put the method below the upload to s3.
If we update the DB record, but something with S3 upload breaks we're gonna end up having a non-existing avatar stored in the DB.